portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -32,66 +31,42 @@
-        } catch(NumberFormatException e) {
-            // Default number of comments
-            numComments = 5;
-        }
-        Query query = new Query("Comments");
-        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-        PreparedQuery results = datastore.prepare(query);
-        
-        ArrayList<Comment> userComments = new ArrayList<>();
-        for(Entity entity: results.asIterable()) {
-            long id = entity.getKey().getId();
-            String username = (String) entity.getProperty("name");
-            String comment = (String) entity.getProperty("comment");
-            
-            Comment userComment =  new Comment(id, username, comment);
-            
-            // Do not exceed max number of comments to display
-            if(userComments.size() >= numComments) {
-                break;
-            }
-
-            userComments.add(userComment);
-        }
-        
-        response.setContentType("application/json");
-
-        //Convert the arraylist to json string
-        String comments = new Gson().toJson(userComments);
-        response.getWriter().println(comments);
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      
+    int numComments;
+    try {
+      numComments = Integer.parseInt(request.getParameter("numComments"));
+    } catch(NumberFormatException e) {
+      // Default number of comments
+      numComments = 5;
     }
 
-    @Override
-    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        String comment = getParameter(request,"comment","");
-        String username = getParameter(request, "name","user");
-        // Do not store an empty comment
-        if(comment.length() == 0) {
-            response.sendRedirect("/contact.html");
-            return;
-        }
-        Entity commentEntity = new Entity("Comments");
-        commentEntity.setProperty("name",username);
-        commentEntity.setProperty("comment",comment);
-        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-        datastore.put(commentEntity);
+    Query query = new Query("Comments");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    
+    ArrayList<Comment> userComments = new ArrayList<>();
+    for(Entity entity: results.asIterable()) {
+      long id = entity.getKey().getId();
+      String username = (String) entity.getProperty("name");
+      String comment = (String) entity.getProperty("comment");
+      String imageUrl = (String) entity.getProperty("imageUrl");
+      
+      Comment userComment =  new Comment(id, username, comment, imageUrl);
+      
+      // Do not exceed max number of comments to display
+      if(userComments.size() >= numComments) {
+        break;
+      }
```

Comment:
Yeah, this is for displaying the number of comments that the user has specified, for instance if they want the numComments to be 0, then I return an empty json at the end since the array would be empty

> Seems like we could do this check and break out of the loop before even reading entity and creating a new Comment

 Yeah, this is for displaying the number of comments that the user has specified, for instance if they want the numComments to be 0, then I return an empty json at the end since the array would be empty